### PR TITLE
Jean S Ariza

### PR DESCRIPTION
**Pull request by jesuarva**
# Original Front APP Trello: There are the latest commits.
- https://trello.com/c/gt45IoLd
- https://trello.com/c/XQpGyNkk

# New Repo for the Front-APP:
I was facing porblems deploying the original Repo, so, literally I copy the core files to a new folder, then I init a new github and local repo, installe dhte dependencies adn lastly deployed to Natlify -> Those acitons solved the issue deploying.
I am still confused to what happened with the first Repo
- https://github.com/jesuarva/lambda-debug

# Trello Boards:
- APP: https://trello.com/b/ntqeFPwr/lambda-notes-jean-s-ariza
- API: https://trello.com/b/6W18iHpl/lambda-notes-backend-jean-s-ariza

### CODE REVIEW

File: auth/loign.router.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const express = require('express');
+const utils = require('./utils');
+const Users = require('../models/Users.model');
+const { RouterFactory } = require('express-router-factory');
```
Comment (nphillips78) [inline auth/loign.router.js]: Aren't you glad you took the time on the front end project to build some reusable elements?

File: auth/utils.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  userHasToken: function(req, res, next) {
+    const token = req.headers.authorization;
+    if (!token) res.status(401).json('You must be registered/logged in before go on');
+    jwt.verify(token, jwk_secret, (err, tokenDecoded) => {
+      req.token = token;
+      req.decodedToken = tokenDecoded;
+      err ? res.status(500).json({ Error: 'Oh ohh, We can not validate your credentials, try again!' }) : next();
+    });
+  },
+};
```
Comment (nphillips78) [inline auth/utils.js]: I like this!

File: db/db.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  connectTo: function(database = process.env.DATABASE || 'sandbox', host = process.env.DB_HOST || 'localhost') {
+    // if there are variables in the .env connecto to that Mongo instance, else connect to localhost instance.
+    return process.env.DB_MONGO_CLUSTER
+      ? mongoose.connect(process.env.DB_MONGO_CLUSTER)
+      : mongoose.connect(`mongodb://${host}/${database}`);
+  },
+  disconnect: function() {
+    return mongoose.disconnect();
+  },
+};
```
Comment (nphillips78) [inline db/db.js]: Beautiful

**Review by nphillips78 — COMMENTED** [review]
Looks great so far! Let me know if you have any issues.

File: auth/loign.router.js (hunk 1 above)
Comment (jesuarva) [reply auth/loign.router.js]: yes, very happy.
Did you noticed that I published it to npm? I use it as any other regular package dependency.
https://www.npmjs.com/package/express-router-factory

I still need to add a readme and instructions to the package in npm.

**Pull request closed without merging**